File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if m*n != len(original):
            return []
        
        result = [[0 for _ in range(n)] for _ in range(m)]
        
        for i,ele in enumerate(original):
            result[i//n][i%n] = ele
            
        return result

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& nums, int m, int n) {
        if(m*n != nums.size())
            return vector<vector<int>>{};
        
        if(m==1 and n==nums.size())
            return vector<vector<int>>{nums};
        vector<vector<int>> res(m,vector<int>(n,0));
        
        for(int i=0;i<nums.size();i++){
            // cout<<i<<" "<<i/m<<" "<<i%n<<endl;
            res[i/n][i%n] = nums[i];
        }
        return res; 
    }
};